### DISCUSSION TITLE
Consider dropping YAML 1.0 support

### DISCUSSION THREAD

**Discussion by eemeli** [eemeli/yaml#200] [General]
For the v2 release, I'm thinking of dropping support for YAML 1.0. If this would inconvenience you, please let me know by commenting here.

**Comment by fkirc** [comment]
For `attranslate`, I would prefer to not drop YAML 1.0: https://github.com/fkirc/attranslate

In the world of translations, all kinds of ancient and unexpected files can occur.
The aim of `attranslate` is to take any YML-file and "just work".

**Comment by eemeli** [comment]
@fkirc Have you encountered any actual YAML 1.0 file that would not properly parse using a later version of the spec? I've collected a list of the differences here: https://eemeli.org/yaml/#version-differences

**Comment by fkirc** [comment]
Until now, I did not know the differences between those versions, therefore it is hard for me to give a definite answer right now.

Nevertheless, I would propose to keep YAML 1.0 in a "best effort" support:

- Retain compatibility in cases where a breakage would only lead to minor code-simplifications.
- Only break compatibility for sub-features that actually complicate internals of the YAML-parser (as opposed to just a few more lines of code).

**Comment by eemeli** [comment]
For some context, YAML 1.0 came out in January 2004, and YAML 1.1 in January 2005. Most simple "YAML 1.0" documents will parse fine with any YAML version, it's only in some of the deeper details that differences emerge.

And I do think that in order to continue supporting a spec that's now 15 years out of date, a bare minimum would be to identify at least one user with one actual YAML 1.0 document that requires some special consideration. I don't really know what else might be required, but that at the very least.

**Comment by fkirc** [comment]
> For some context, YAML 1.0 came out in January 2004, and YAML 1.1 in January 2005. Most simple "YAML 1.0" documents will parse fine with any YAML version, it's only in some of the deeper details that differences emerge.

This is exactly the thing that I meant with "best effort support". On the one hand, I hope that most simple "YAML 1.0" documents will continue to work in the future.
On the other hand, I fully understand if you drop support for arcane features that are hard to maintain and hard to test.
